### PR TITLE
Fix OOM error when processing Full Scene

### PR DESCRIPTION
**Pull request by rodrigoalmeida94**
- [x] Add some missing fstrings in logs

- [x] Add gc magic to manage memory

### CODE REVIEW

File: src/inference.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ def save_result(
 
     with rasterio.open(image_name, "w", **output_profile) as d_s:
         for b_i, b_n in enumerate(output_bands):
-            d_s.write(model_output[:, :, b_i].astype(np.uint16), indexes=b_i + 1)
+            d_s.write(model_output[:, :, b_i], indexes=b_i + 1)
```
Comment (markusuwe) [inline src/inference.py:40]: Does this have an actual effect or was it just unnecessary?

File: src/supres.py [hunk 2]
```diff
@@ -1,5 +1,6 @@
 from __future__ import division
 
+import gc
```
Comment (markusuwe) [inline src/supres.py:3]: So it is imported, but not used but it helps?

File: src/supres.py [hunk 3]
```diff
@@ -41,6 +42,7 @@ def dsen2_20(d10, d20, image_level):
         model_filename = L2A_MDL_PATH_20M_DSEN2
 
     prediction = _predict(test, model_filename)
+    del test, p10, p20
```
Comment (markusuwe) [inline src/supres.py:45]: I assume these are the important changes in the context of this PR?

**Review by markusuwe — APPROVED** [review]
LGTM
Got a few questions.

File: src/supres.py (hunk 2 above)
Comment (rodrigoalmeida94) [reply src/supres.py:3]: gc module is used in L115

File: src/supres.py (hunk 3 above)
Comment (rodrigoalmeida94) [reply src/supres.py:45]: Yes, deleting "not required anymore" variables is a part of it in conjunction with gc.collect().

File: src/inference.py (hunk 1 above)
Comment (rodrigoalmeida94) [reply src/inference.py:40]: It did have an effect, but that was when the incoming data was a float, now I'm doing the casting to uint16 in L111-117

File: src/supres.py (hunk 2 above)
Comment (markusuwe) [reply src/supres.py:3]: Thanks. Overlooked it as it is part of a logger statement. Interesting way to invoke gc.

**Review by marooned20 — COMMENTED** [review]
I am curious of the performance with respect to all the deleted vars and type-casting with and without gc.
Is there a difference in the performance?

**Review by nekhtiari — APPROVED** [review]
Amazing!

**Comment by rodrigoalmeida94** [comment]
> I am curious of the performance with respect to all the deleted vars and type-casting with and without gc.
> Is there a difference in the performance?

What seems to do the trick in terms of memory is the combination of gc and delete of large numpy arrays. The type casting location I figure is not really relevant.

**Review by chrieke — COMMENTED** [review]
LGTM

**Pull request merged**